hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecordPayload.java

```diff
@@ -162,6 +162,23 @@ default Comparable<?> getOrderingValue() {
     return 0;
   }
 
+  /**
+   * Returns true is this payload represents an actual record to be written.
+   *
+   * Default implementation checks if the insert value is present. For better write performance, this should be
+   * overridden by implementations for which generating the insert value is expensive.
+   *
+   * @param schema Schema used for record
+   * @param properties Payload related properties. For example pass the ordering field(s) name to extract from value in storage.
+   * @return true if this payload has a record to be written.
+   *         false if this payload is empty and should be skipped from writing.
+   *         false if this payload represents a record to be deleted.
+   */
+  @PublicAPIMethod(maturity = ApiMaturityLevel.EVOLVING)
+  default boolean hasInsertValue(Schema schema, Properties properties) throws IOException {
```

Comment:
Payloads have been deprecated in favor of merge modes which have similar optimizations made. Is it possible for your workflow to adopt a custom merger or one of the prebuilt merge modes?

Reply:
yeah, probably put the patch to just 0.x-branch.